File: react-native/bridge/native_module_sample/ios/CalendarManager.m
// CalendarManager.m
#import "CalendarManager.h"
#import <React/RCTLog.h>

@implementation CalendarManager

// To export a module named CalendarManager
RCT_EXPORT_MODULE();

// This would name the module AwesomeCalendarManager instead
// RCT_EXPORT_MODULE(AwesomeCalendarManager);

RCT_EXPORT_METHOD(addEvent:(NSString *)name location:(NSString *)location)
{
  RCTLogInfo(@"Pretending to create an event %@ at %@", name, location);
}

@end

File: react-native/bridge/native_module_sample/android/app/src/main/java/com/native_module_sample/CalendarManager.java
package com.native_module_sample;

import android.nfc.Tag;
import android.util.Log;

import androidx.annotation.NonNull;

import com.facebook.react.bridge.NativeModule;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.facebook.react.bridge.ReactMethod;

public class CalendarManager extends ReactContextBaseJavaModule {
    private static final String TAG = "CalendarManager";
    private static ReactApplicationContext reactContext;

    public CalendarManager(@NonNull ReactApplicationContext reactContext) {
        super(reactContext);
    }

    @NonNull
    @Override
    public String getName() {
        return "CalendarManager";
    }

    @ReactMethod
    public void addEvent(String name, String location) {
        Log.d(TAG, String.format("Pretending to create an event %s at %s", name, location));
    }
}
